src/app/api/users/users.service.ts

```diff
@@ -0,0 +1,30 @@
+import { CopilotAPI } from '@/utils/CopilotAPI'
+import { BaseService } from '@api/core/services/base.service'
+import User from '@api/core/models/User.model'
+import { PoliciesService } from '@api/core/services/policies.service'
+import { Resource } from '@api/core/types/api'
+import { UserAction } from '@api/core/types/user'
+
+class UsersService extends BaseService {
+  private copilot: CopilotAPI
+
+  constructor(user: User) {
+    super(user)
+    this.copilot = new CopilotAPI(this.user.token)
+  }
+
+  async getGroupedUsers() {
+    const user = this.user
+    new PoliciesService(user).authorize(UserAction.Read, Resource.Users)
```

Comment:
I don't see the purpose this line of code is serving. If you removed the line of code what would not work?

Reply:
This is meant to be resource-level authorization for the service layer.

Let's say we query the `/api/users` endpoint as a client - then we can access a list of all the IUs, companies that this organization is working with, as well as all of their clients!

The PoliciesService validates that the current `user` is authorized to read (`UserAction.Read`) the users resource (`Resource.Users` in this case. @makeitraina

Reply:
It's in the `policies.service.ts` file. We are using a hash lookup right now @makeitraina 

![Screenshot from 2024-04-30 10-07-23](https://github.com/copilot-platforms/tasks-app/assets/59971845/63c69f5e-a538-4601-b8a9-3eb3db3421d3)